legal-api/src/legal_api/models/batch_processing.py

```diff
@@ -0,0 +1,96 @@
+from legal_api.utils.datetime import datetime
+
+from .db import db
+
+
+class BatchProcessing(db.Model):  # pylint: disable=too-many-instance-attributes
+    """This class manages the batch processing."""
+
+    class BatchProcessingStep(BaseEnum):
+        """Render an Enum of the batch processing step."""
+
+        WARNING_LEVEL_1 = auto()
+        WARNING_LEVEL_2 = auto()
+        DISSOLUTION = auto()
+
+    class BatchProcessingStatus(BaseEnum):
+        """Render an Enum of the batch processing status."""
+
+        HOLD = auto()
+        PROCESSING = auto()
+        WITHDRAWN = auto()
+        COMPLETED = auto()
+        ERROR = auto()
+
+    __tablename__ = 'batch_processing'
+
+    id = db.Column(db.Integer, primary_key=True)
+    business_identifier = db.Column('business_identifier', db.String(10), default='', nullable=False)
+    step = db.Column('step', db.Enum(BatchProcessingStep), nullable=False)
+    status = db.Column('status', db.Enum(BatchProcessingStatus), nullable=False)
+    notes = db.Column('notes', db.String(150), default='', nullable=True)
+    created_date = db.Column('created_date', db.DateTime(timezone=True), default=datetime.utcnow)
+    last_modified = db.Column('last_modified', db.DateTime(timezone=True), default=datetime.utcnow)
+
+    # parent keys
+    batch_id = db.Column('batch_id', db.Integer, db.ForeignKey('batches.id'), index=True, nullable=False)
+    business_id = db.Column('business_id', db.Integer, db.ForeignKey('businesses.id'), index=True, nullable=False)
+
+    def save(self):
+        """Save the object to the database immediately."""
+        db.session.add(self)
+        db.session.commit()
+
+    @classmethod
+    def find_by_id(cls, batch_processing_id: int):
+        """Return the batch matching the id."""
+        batch_processing = None
+        if batch_processing_id:
+            batch_processing = cls.query.filter_by(id=batch_processing_id).one_or_none()
+        return batch_processing
+
+    @classmethod
+    def find_by(cls,  # pylint: disable=too-many-arguments
+                batch_id: int = None,
+                business_id: int = None,
+                step: BatchProcessingStep = None,
+                status: BatchProcessingStatus = None,
+                notes: str = None) -> dict:
+        """Return the batch matching."""
+        query = db.session.query(BatchProcessing)
+        batch_processinges = []
+
+        if batch_id:
+            query = query.filter(BatchProcessing.batch_id == batch_id)
+
+        if business_id:
+            query = query.filter(BatchProcessing.business_id == business_id)
+
+        if step:
+            query = query.filter(BatchProcessing.step == step)
+
+        if status:
+            query = query.filter(BatchProcessing.status == status)
+
+        if notes:
+            query = query.filter(BatchProcessing.notes == notes)
+
+        batch_processinges = query.order_by(BatchProcessing.id).all()
+        return batch_processinges
```

Comment:
can we remove `notes` for now?  Don't see a use case for searching by that yet

Reply:
Removed